Build/Build.cs

```diff
@@ -32,17 +70,19 @@ class Build : NukeBuild
 
     public static int Main() => Execute<Build>(x => x.SpellCheck, x => x.Push);
 
-    [Parameter("A branch specification such as develop or refs/pull/1775/merge")]
-    readonly string BranchSpec;
+    GitHubActions GitHubActions => GitHubActions.Instance;
 
-    [Parameter("An incrementing build number as provided by the build engine")]
-    readonly string BuildNumber;
-
-    [Parameter("The target branch for the pull request")]
-    readonly string PullRequestBase;
+    string BranchSpec => GitHubActions?.Ref;
+    string BuildNumber => GitHubActions?.RunNumber.ToString();
+    string PullRequestBase => GitHubActions?.BaseRef;
 
     [Parameter("The key to push to Nuget")]
-    readonly string ApiKey;
+    [Secret]
+    readonly string Nugetapikey;
```

Comment:
If this PR should be merged you have to rename the github secret `NUGETAPIKEY` either to `NUGET_API_KEY`or `API_KEY` to get the correct spelling when generating the yaml.